dipy/viz/ui.py

```diff
@@ -0,0 +1,612 @@
+
+    def add_to_renderer(self, ren):
+        """ Allows UI objects to add their own props to the renderer.
+        Here, we add only call add_to_renderer for the additional components.
+
+        Parameters
+        ----------
+        ren : renderer
+        """
+        super(Panel2D, self).add_to_renderer(ren)
+        for ui_item in self.ui_list:
+            ui_item.add_to_renderer(ren)
+
+    def get_actors(self):
+        """ Returns the panel actor. """
+        return [self.panel.actor]
+
+    def add_callback(self, event_type, callback):
+        """ Adds events to an actor.
+
+        Parameters
+        ----------
+        event_type : string
+            event code
+        callback : function
+            callback function
+        """
+        super(Panel2D, self).add_callback(self.panel.actor, event_type, callback)
+
+    def add_element(self, element, position_type, position):
+        """ Adds an element to the panel.
+        The center of the rectangular panel is its bottom lower position.
+
+        Parameters
+        ----------
+        element : UI
+            The UI item to be added.
+        position_type: string
+            'absolute' or 'relative'
+        position : (float, float)
+            Absolute for absolute and relative for relative
+        """
+        self.ui_list.append(element)
+        if position_type == 'relative':
+            self.element_positions.append([element, position_type, position[0], position[1]])
+            element.set_center((self.lower_limits[0] + position[0] * self.size[0],
+                                self.lower_limits[1] + position[1] * self.size[1]))
+        elif position_type == 'absolute':
+            self.element_positions.append([element, position_type, position[0], position[1]])
+            element.set_center((position[0], position[1]))
+        else:
+            raise ValueError("Position can only be absolute or relative")
+
+    def set_center(self, position):
+        """ Sets the panel center to position.
+        The center of the rectangular panel is its bottom lower position.
+
+        Parameters
+        ----------
+        position : (float, float)
+        """
+        shift = [position[0] - self.center[0], position[1] - self.center[1]]
+        self.center = position
+        self.lower_limits = (position[0] - self.size[0] / 2, position[1] - self.size[1] / 2)
+        for ui_element in self.element_positions:
+            if ui_element[1] == 'relative':
+                ui_element[0].set_center((self.lower_limits[0] + ui_element[2] * self.size[0],
+                                          self.lower_limits[1] + ui_element[3] * self.size[1]))
+            elif ui_element[1] == 'absolute':
+                ui_element[2] += shift[0]
+                ui_element[3] += shift[1]
+                ui_element[0].set_center((ui_element[2], ui_element[3]))
+
+    @staticmethod
+    def left_button_press(i_ren, obj, element):
```

Comment:
`element` here is expected to be a `Panel2D` object? Maybe change the name to something more precise like `panel2d_object`.